network/eos/eos_eapi.py

```diff
@@ -0,0 +1,280 @@
+            return result
+
+    if config.get('enabled') and state == 'stopped':
+        # turn off eapi access and exit
+        commands.append('shutdown')
+        result['changed'] = True
+        result['commands'] = commands
+        execute_commands(module, commands)
+        return result
+
+    # http and https
+    for protocol in ['http', 'https']:
+        cmds, chg = http_commands(protocol, module.params['{0}_port'.format(protocol)],
+                                  module.params['{0}'.format(protocol)], config)
+        if chg:
+            commands += cmds
+            result['changed'] = True
+
+    # local HTTP
+    if config.get('localHttpServer').get('running'):
+        if not local_http:
+            # turn off local http server
+            commands.append('no protocol http localhost')
+            result['changed'] = True
+        elif config.get('localHttpServer').get('port') != local_http_port:
+            # update the local http port
+            commands.append('protocol http localhost port {0}'.format(local_http_port))
+            result['changed'] = True
+
+    if not config.get('localHttpServer').get('running') and local_http:
+        # turn on local http server
+        commands.append('protocol http localhost port {0}'.format(local_http_port))
+        result['changed'] = True
+
+    # socket server
+    if config.get('unixSocketServer').get('running') and not socket:
+        # turn off unix socket
+        commands.append('no protocol unix-socket')
+        result['changed'] = True
+
+    if not config.get('unixSocketServer').get('running') and socket:
+        # turn on unix socket
+        commands.append('protocol unix-socket')
+        result['changed'] = True
+
+    if len(commands) > 1:
+        # something requires change
+        execute_commands(module, commands)
+        result['commands'] = commands
+
+    return result
+
+def check_version(module):
+    config = module.from_json(module.execute(['show version | json'])[0])
+    versions = config['version'].split('.')
+    if int(versions[0]) < 4 or int(versions[1]) < 12:
+        module.fail_json(msg="Device version {0} does not support eAPI. eAPI was introduced in EOS 4.12.")
+
+def main():
+    """ main entry point for module execution
+    """
+
+    argument_spec = dict(
+        state=dict(default='started', choices=['stopped','started']),
+        http_port=dict(default=80, type='int'),
+        https_port=dict(default=443, type='int'),
+        local_http_port=dict(default=8080, type='int'),
+        http=dict(aliases=['enable_http'], default=True, type='bool'),
+        https=dict(aliases=['enable_https'], default=True, type='bool'),
+        socket=dict(aliases=['enable_socket'], default=False, type='bool'),
+        local_http=dict(aliases=['enable_local_http'], default=False, type='bool'),
+
+        # Only allow use of transport cli when coniguring EAPI
+        transport=dict(required=True, choices=['cli'])
```

Comment:
Same comment as nxos_nxapi. Modifying the parameter causes error to be thrown before anything further is evaluated by the interpreter. The resulting error message lets the user know that the issue is a bad transport value and only cli is supported, which seems reasonable enough.